code/modules/games/cards.dm

```diff
@@ -0,0 +1,435 @@
+	else
+		user.visible_message("\The [user] deals [dcard] card(s) to \the [target].")
+	H.throw_at(get_step(target,target.dir),10,1,H)
+
+
+/obj/item/weapon/hand/attackby(obj/O as obj, mob/user as mob)
+	if(cards.len == 1 && istype(O, /obj/item/weapon/pen))
+		var/datum/playingcard/P = cards[1]
+		if(P.name != "Blank Card")
+			to_chat(user,"You cannot write on that card.")
+			return
+		var/cardtext = sanitize(input(user, "What do you wish to write on the card?", "Card Editing") as text|null, MAX_PAPER_MESSAGE_LEN)
+		if(!cardtext)
+			return
+		P.name = cardtext
+		// SNOWFLAKE FOR CAG, REMOVE IF OTHER CARDS ARE ADDED THAT USE THIS.
+		P.card_icon = "cag_white_card"
+		update_icon()
+	else if(istype(O,/obj/item/weapon/hand))
+		var/obj/item/weapon/hand/H = O
+		for(var/datum/playingcard/P in cards)
+			H.cards += P
+		H.concealed = src.concealed
+		qdel(src)
+		H.update_icon()
+		return
+	..()
+
+/obj/item/weapon/deck/attack_self()
+	shuffle()
+
+/obj/item/weapon/deck/verb/verb_shuffle()
+	set category = "Object"
+	set name = "Shuffle"
+	set desc = "Shuffle the cards in the deck."
+	set src in view(1)
+	shuffle()
+
+/obj/item/weapon/deck/proc/shuffle()
+	var/mob/living/user = usr
+	if (cooldown < world.time - 15) // 15 ticks cooldown
+		var/list/newcards = list()
+		while(cards.len)
+			var/datum/playingcard/P = pick(cards)
+			newcards += P
+			cards -= P
+		cards = newcards
+		user.visible_message("\The [user] shuffles [src].")
+		playsound(user, 'sound/items/cardshuffle.ogg', 50, 1)
+		cooldown = world.time
+	else
+		return
+
+
+/obj/item/weapon/deck/MouseDrop(atom/over_object) // Code from Paper bin, so you can still pick up the deck
+	var/mob/M = usr
+	if(M.restrained() || M.stat || !Adjacent(M))
+		return
+
+	if(over_object == M)
+		M.put_in_hands(src)
+
+	else if(istype(over_object, /obj/screen))
+		switch(over_object.name)
+			if("r_hand")
+				if(!remove_item_from_storage(M))
+					M.unEquip(src)
+				M.put_in_r_hand(src)
+			if("l_hand")
+				if(!remove_item_from_storage(M))
+					M.unEquip(src)
+				M.put_in_l_hand(src)
+
+	add_fingerprint(M)
+
+	return
+
+/obj/item/weapon/pack/
+	name = "Card Pack"
+	desc = "For those with disposible income."
+
+	icon_state = "card_pack"
+	icon = 'icons/obj/playing_cards.dmi'
+	w_class = 1
+	var/list/cards = list()
+
+
+/obj/item/weapon/pack/attack_self(var/mob/user as mob)
+	user.visible_message("[user] rips open \the [src]!")
+	var/obj/item/weapon/hand/H = new()
+
+	H.cards += cards
+	cards.Cut();
+	user.drop_item()
+	qdel(src)
+
+	H.update_icon()
+	user.put_in_active_hand(H)
+
+/obj/item/weapon/hand
+	name = "hand of cards"
+	desc = "Some playing cards."
+	icon = 'icons/obj/playing_cards.dmi'
+	icon_state = "empty"
+	w_class = 1
+
+	var/concealed = 0
+	var/list/cards = list()
+
+/obj/item/weapon/hand/verb/discard()
+
+	set category = "Object"
+	set name = "Discard"
+	set desc = "Place (a) card(s) from your hand in front of you."
```

Comment:
Would probably just be better as `"Place cards from your hand in front of you."`